broccoli-image-writer.js

```diff
@@ -29,39 +27,32 @@ class ImageResizer extends CachingWriter {
     }
   }
 
-  writeRed(message) {
-    if (this.ui) {
-      this.ui.writeLine(chalk.red(message));
-    }
-  }
-
   build() {
     let sourcePath = this.inputPaths[0];
     let destinationPath = path.join(this.outputPath, this.image_options.destinationDir);
-    let promises = [];
-    try {
-      //make destination folder, if not exists
-      mkdirp.sync(destinationPath);
-      //read files from source folder
-      let files = fs.readdirSync(sourcePath);
-      files.forEach((file) => {
-        this.writeGreen(file);
-
-        let newPromise;
-        if (this.image_options.justCopy) {
-          newPromise = this.copyImages(file, sourcePath, destinationPath);
-        } else {
-          newPromise = this.generateImages(file, sourcePath, destinationPath);
-        }
-        promises = promises.concat(newPromise);
+    let justCopy = this.image_options.justCopy;
+
+    //make destination folder, if not exists
+    fs.ensureDirSync(destinationPath);
+    //read files from source folder
+    let files = fs.readdirSync(sourcePath);
+
+    let tasks = files
+      .map((file) => {
+        // return a promise returning function
+        return () => {
+          this.writeGreen(file);
+
+          if (justCopy) {
+            return this.copyImages(file, sourcePath, destinationPath);
+          } else {
+            return this.generateImages(file, sourcePath, destinationPath);
+          }
+        };
       });
-    } catch (e) {
-      this.writeRed(e);
```

Comment:
@andreasschacht I was unsure about this `catch` here. Do we need to catch? Or should Ember CLI just catch the error and display it?

Reply:
I don't remember. I think, we don't need it anymore.